### PR TITLE
[koa-logger] fix type for transporter args parameter

### PR DESCRIPTION
**Pull request by ian-rudge**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/koajs/logger/blob/f8edfa00cb5af7e696cf276ddc8b482accd9f7a9/index.js#L33
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/koa-logger/index.d.ts [hunk 1]
```diff
@@ -3,18 +3,19 @@
 // Definitions by: Joshua DeVinney <https://github.com/geoffreak>
 //                 Tomek Łaziuk <https://github.com/tlaziuk>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// Minimum TypeScript Version: 4.0
```
Comment (peterblazejewicz) [inline types/koa-logger/index.d.ts:6]: `types/koa` has `// TypeScript Version: 3.0` itself.
what's is here to gain really by using named tuples?
If this provides some useful gains, you'd probably have to 
split implementation for <=4.0:
https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-very-new-typescript-versions

**Review by andrewbranch — CHANGES_REQUESTED** [review]
The named tuple elements do seem nice for intellisense/documentation purposes, but I don’t think it’s worth breaking people on older TS versions over, nor do I think it’s worth splitting the definitions up to support multiple TS version ranges. I would suggest just removing the tuple element names, bumping the minimum TS version back down, and revisiting this in about a year when DT’s minimum supported TS version will be closer to 4.0 across the board.

**Comment by ian-rudge** [comment]
> The named tuple elements do seem nice for intellisense/documentation purposes, but I don’t think it’s worth breaking people on older TS versions over, nor do I think it’s worth splitting the definitions up to support multiple TS version ranges. I would suggest just removing the tuple element names, bumping the minimum TS version back down, and revisiting this in about a year when DT’s minimum supported TS version will be closer to 4.0 across the board.

Yea it is really nice in this case but I agree, probably just worth fixing the type and keeping it compatible.

File: types/koa-logger/index.d.ts (hunk 1 above)
Comment (ian-rudge) [reply types/koa-logger/index.d.ts]: It is nicer to have named tuples as it provides a better typing experience, however as mentioned below, maybe it's worth just fixing the type for now

**Comment by andrewbranch** [comment]
The lowest supported TS version on DefinitelyTyped _does_ support tuples, just not _names_ on tuple elements. You could still capture the types of which elements are which, though I can’t easily tell from the source code that what you have is accurate.

**Comment by andrewbranch** [comment]
@ian-rudge can you point out somewhere in the docs or source code that validates the tuple type you’ve written?

**Comment by ian-rudge** [comment]
> @ian-rudge can you point out somewhere in the docs or source code that validates the tuple type you’ve written?

looks like there's a couple places it's called, first on the request start you can see the parameters passed to the function: https://github.com/koajs/logger/blob/f8edfa00cb5af7e696cf276ddc8b482accd9f7a9/index.js#L44

then on the request end we get a few more parameters passed to the function: https://github.com/koajs/logger/blob/f8edfa00cb5af7e696cf276ddc8b482accd9f7a9/index.js#L118

also in the [readme](https://github.com/koajs/logger/blob/master/README.md) if you scroll to the bottom it mentions: "Param args is a array by `[format, method, url, status, time, length]`"

**Review by andrewbranch — APPROVED** [review]

**Comment by ian-rudge** [comment]
Ready to merge

**Pull request merged**